CMakeLists.txt

```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.20)
 
 set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# Enables symbol export for loadable modules, ensuring same shared resource reference.
+set(CMAKE_ENABLE_EXPORTS ON)
```

Comment:
Please consider this change:
```
set(CMAKE_CXX_STANDARD 23)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

# Enable symbol export for loadable modules, ensuring same shared resource
# references. Needed for the plugins, as they are shared libraries.
set(CMAKE_ENABLE_EXPORTS ON)
```
IMHO, it explains better why we are introducing this flag. It is also separated from the C++ standard definition.